### PR TITLE
feat: broker requests can address specific brokers

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

Broker requests can optionally provide a broker id to indicate that this request should be sent to a specific broker, regardless of whether this broker is leader for the partition or not.

## Related issues

relates to #9630 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerRequest.java [hunk 1]
```diff
@@ -36,6 +37,8 @@ public BrokerRequest(final int schemaId, final int templateId) {
     this.templateId = templateId;
   }
 
+  public abstract Optional<Integer> addressesSpecificBroker();
```
Comment (deepthidevaki) [inline gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerRequest.java:40]: :wrench: I prefer to have the following interface
```
boolean addressSpecificBroker();
int getBorkerId();
```

You can already give default implementations of this in `BrokerRequest`. So only the requests that need to address specificBroker have to override the implementations.

**Review by deepthidevaki — COMMENTED** [review]
Please see my comment and let me know what you think of the suggestion. I will do the review after your reply.

File: gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerRequest.java (hunk 1 above)
Comment (oleschoenburg) [reply gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerRequest.java:40]: That's how I started (it's the same for addressing partitions) but then I couldn't find any reason to not just have a single method for this. I can still give a default implementation in BrokerRequest that just returns `Optional.empty()`

Is there a specific reason you prefer that interface?

Comment (deepthidevaki) [reply gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerRequest.java:40]: I don't have a strong opinion on this. If only one method then may be rename `addressSpecificBroker` to something like `getBroker`?

Default implementation would be good either way.

Comment (oleschoenburg) [reply gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerRequest.java]: @deepthidevaki Updated the method name and added a default implementation

**Review by deepthidevaki — APPROVED** [review]
Thanks :+1:

**Pull request merged**